packages/itwinui-css/src/button/button.scss

```diff
@@ -22,49 +19,99 @@ $iui-button-padding-large: $iui-xs * 6;
   position: relative;
   border-radius: $iui-border-radius;
   line-height: $iui-line-height;
-  box-shadow: none;
   font-size: $iui-font-size;
   font-weight: $iui-font-weight-normal;
-  text-decoration: none;
   user-select: none;
   cursor: pointer;
   white-space: nowrap;
-  border: 1px solid transparent;
+  border: 1px solid var(--_iui-button-border-color);
+  background-color: var(--_iui-button-background-color);
   color: var(--_iui-button-text-color);
-  @media (prefers-reduced-motion: no-preference) {
-    transition: color $iui-speed-fast ease-out, background-color $iui-speed-fast ease-out,
-      border-color $iui-speed-fast ease-out;
+  transition: background-color $iui-speed-fast ease-out, border-color $iui-speed-fast ease-out;
+  -webkit-tap-highlight-color: transparent;
+
+  @at-root a#{&} {
+    text-decoration: none;
   }
 
-  &:hover {
-    --_iui-button-text-color: rgba(var(--iui-color-foreground-body-rgb), var(--iui-opacity-1));
+  // #region Sizing
+  @include iui-button-size;
 
-    text-decoration: none;
+  &[data-iui-button-size='small'] {
+    @include iui-button-size(small);
   }
 
-  @include iui-focus;
+  &[data-iui-button-size='large'] {
+    @include iui-button-size(large);
+  }
+  // #endregion Sizing
+
+  &[aria-disabled='true'] {
+    @include iui-focus;
+  }
 
   &[disabled],
-  &:disabled {
-    @include iui-button-disabled;
+  &:disabled,
+  &[aria-disabled='true'] {
+    cursor: not-allowed;
+
+    &,
+    &:hover,
+    &:active {
```

Comment:
Is there any reason to add all these extra selectors?

Ideally, disabled always comes at the end and overrides any previous states. If specificity is an issue, maybe we should use `:where` in all the places that we change `:hover` states.

Reply:
Can you explain how I can use `:where` to fix this? I'm not understanding.